samcli/commands/local/lib/local_function_url_service.py

```diff
@@ -0,0 +1,351 @@
+
+        Parameters
+        ----------
+        port : int
+            Port number to check
+
+        Returns
+        -------
+        bool
+            True if port is available, False otherwise
+        """
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                sock.bind((self.host, port))
+                return True
+        except OSError:
+            LOG.debug(f"Port {port} is already in use")
+            return False
+
+    def _start_function_service(self, func_name: str, func_config: Dict, port: int) -> FunctionUrlHandler:
+        """Start individual function URL service"""
+        service = FunctionUrlHandler(
+            function_name=func_name,
+            function_config=func_config,
+            local_lambda_runner=self.invoke_context.local_lambda_runner,
+            port=port,
+            host=self.host,
+            disable_authorizer=self.disable_authorizer,
+            stderr=self.invoke_context.stderr,
+            ssl_context=None,
+        )
+        return service
+
+    def start(self):
+        """
+        Start the Function URL services. This method will block until stopped.
+        """
+        if not self.function_urls:
+            raise NoFunctionUrlsDefined("No Function URLs found to start")
+
+        # Setup signal handlers
+        def signal_handler(sig, frame):
+            LOG.info("Received interrupt signal. Shutting down...")
+            self._shutdown_event.set()
+
+        signal.signal(signal.SIGINT, signal_handler)
+        signal.signal(signal.SIGTERM, signal_handler)
+
+        # Start services
+        self.executor = ThreadPoolExecutor(max_workers=len(self.function_urls))
+
+        try:
+            # Start each function service
+            for func_name, func_config in self.function_urls.items():
+                port = self._allocate_port()
+                service = self._start_function_service(func_name, func_config, port)
+                self.services[func_name] = service
+
+                # Start the service (this runs Flask in a thread)
+                service.start()
+
+                # Wait for the service to be ready
+                if not self._wait_for_service(port):
+                    LOG.warning(f"Service for {func_name} on port {port} did not start properly")
+
+            # Print startup info
+            self._print_startup_info()
+
+            # Wait for shutdown signal
+            self._shutdown_event.wait()
+
+        except KeyboardInterrupt:
+            LOG.info("Received keyboard interrupt")
+        finally:
+            self._shutdown_services()
+
+    def start_all(self):
+        """
+        Start all Function URL services. Alias for start() method.
+        """
+        return self.start()
+
+    def start_function(self, function_name: str, port: int):
```

Comment:
I feel like we shouldn't have two different functions for `start` and `start_function`. Both are basically the same thing, but with one function vs multiple. We shouldn't have to duplicate the code.